review update-version script

This script is causing problems with web-ext. Do we still need it?

FWIW, this might also be something to consider an upstream issue, if we consider this to be a valid version format for our own self-hosted add-on. (We basically do consider it valid, since we've been doing it this way for a very long time. But we could change our thinking on that.)

<strike>Also, I haven't seen what the exact error is yet - is it an *error* that fails the build? a noisy *warning* we can ignore / filter? something we can configure `web-ext lint` to ignore?</strike>

Okay, looks like this is exactly what we get from `npm run lint:extension`:
```
C:\Users\lmorchard\devel\testpilot\addon [3111-more-webpack +5 ~1 -0 !]
λ  npm run lint:extension

> testpilot-addon@3.0.0-dev-c4ce10c lint:extension C:\Users\lmorchard\devel\testpilot\addon
> web-ext lint --self-hosted

Applying config file: .\package.json
Validation Summary:

errors          1
notices         0
warnings        2

ERRORS:

Code           Message             Description                           File   Line   Column
JSON_INVALID   "/version" should   Your JSON file could not be parsed.
               match format
               "versionString"
WARNINGS:

Code             Message                Description                                  File                     Line   Column
DANGEROUS_EVAL   The Function           Evaluation of strings as code can lead to    build/webextension/ba…   727    11

                 constructor is eval.   security vulnerabilities and performance     ckground.js

                                        issues, even in the most innocuous of

                                        circumstances. Please avoid using `eval`

                                        and the `Function` constructor when at all

                                        possible.'

DANGEROUS_EVAL   eval can be harmful.   Evaluation of strings as code can lead to    build/webextension/ba…   727    43

                                        security vulnerabilities and performance     ckground.js

                                        issues, even in the most innocuous of

                                        circumstances. Please avoid using `eval`

                                        and the `Function` constructor when at all

                                        possible.'

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! testpilot-addon@3.0.0-dev-c4ce10c lint:extension: `web-ext lint --self-hosted`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the testpilot-addon@3.0.0-dev-c4ce10c lint:extension script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\lmorchard\AppData\Roaming\npm-cache\_logs\2018-04-12T16_44_23_418Z-debug.log
```

--- a/addon/bin/update-version
+++ b/addon/bin/update-version
@@ -1,29 +1,24 @@
 #!/usr/bin/env node
 
-var fs = require('fs');
-var util = require('util');
-var childProcess = require('child_process');
+const fs = require('fs');
+const util = require('util');
+const childProcess = require('child_process');
 
 // Load the current package.json
-var packageFilename = __dirname + '/../package.json';
-var package = require(packageFilename);
+const packageFilename = __dirname + '/../package.json';
+const package = require(packageFilename);
+
 
 // Come up with a version tag suffix based on CIRCLE_TAG, or fall back to
 // looking up the current git commit hash.
-var tag;
-if (process.env['CIRCLE_TAG']) {
-  tag = 'tag-' + process.env['CIRCLE_TAG'];
-} else {
-  // Note: Must use short git hash, because AMO rejects versions > 32 chars
-  tag = 'dev-' + childProcess.execSync('git rev-parse --short HEAD')
-                             .toString('utf-8')
-                             .trim();
-}
+const tag = 'v' + childProcess
+    .execSync('git rev-parse --short HEAD').toString('utf-8').trim()
+    .split('').sort().reverse().join('');
+
+const semver = /^(\d+\.?){1,3}/.exec(package.version)[0];
 
 // Update the version tag suffix.
-var versionParts = package.version.split('-');
-var origVersion = package.version;
-package.version = versionParts[0] + '-' + tag;
+package.version = semver + tag;
 
 // Write the modified package.json
 var packageJSON = JSON.stringify(package, null, '  ');

--- a/addon/manifest.json
+++ b/addon/manifest.json
@@ -8,7 +8,7 @@
       "update_url": "https://testpilot.firefox.com/files/@testpilot-addon/updates.json"
     }
   },
-  "version": "3.0.0",
+  "version": "3.0.0vffeb6433",
   "icons": {
     "16": "icons/icon-16.png",
     "32": "icons/icon-32.png",

--- a/addon/package.json
+++ b/addon/package.json
@@ -2,7 +2,7 @@
   "title": "Test Pilot",
   "name": "testpilot-addon",
   "id": "@testpilot-addon",
-  "version": "3.0.0",
+  "version": "3.0.0vffeb6433",
   "private": true,
   "description": "Test Pilot is a privacy-sensitive user research program focused on getting new features into Firefox faster.",
   "repository": "mozilla/testpilot",
@@ -22,7 +22,7 @@
   },
   "scripts": {
     "start": "web-ext run",
-    "prepackage": "npm run locale && npm run lint && npm run lint:extension",
+    "prepackage": "npm run locale && npm run lint && npm run lint:extension && npm run update-version",
     "package": "shx rm -rf web-ext-artifacts && cross-env web-ext build --overwrite-dest && shx mv web-ext-artifacts/test_pilot-*.zip ./addon.xpi",
     "lint": "eslint .",
     "lint:extension": "web-ext lint --self-hosted",

